lib/validators/icons-validator.ts

```diff
@@ -22,17 +22,29 @@ export class IconsValidator extends Validator {
     }
 
     async validate(): Promise<void> {
-        for (const component of Object.values(this.componentSet.components)) {
-            const ext = path.extname(component.icon).toLowerCase();
+        for (const comp of Object.values(this.componentSet.components)) {
+            const ext = path.extname(comp.icon).toLowerCase();
             if (this.supportedFormats.indexOf(ext) === -1) {
                 this.error(`Icons are only supported in SVG or transparent PNG format`);
             } else if (ext === '.png') {
-                const data = await this.getFileContent(path.normalize(component.icon));
-                const png = PNG.sync.read(data as Buffer);
+                await this.validatePngIcon(comp.name, comp.icon);
+            }
+        }
+    }
 
-                if (png.alpha === false) {
-                    this.error(`PNG icons are only supported when they are transparent`);
-                }
+    private async validatePngIcon(name: Component['name'], icon: Component['icon']) {
+        try {
+            const data = await this.getFileContent(path.normalize(icon));
+            const png = PNG.sync.read(data as Buffer);
+
+            if (png.alpha === false) {
+                this.error(`PNG icons are only supported when they are transparent`);
+            }
+        } catch (e) {
+            if (e.message === 'Invalid file signature') {
+                this.error(`Component "${name}" icon "${icon}" is not a valid PNG file`);
+            } else {
+                throw e;
```

Comment:
Bit in doubt if we should still throw in this case. Maybe we could also have something like this:
```ts
this.error(`Unable to validate component "${name}" icon "${icon}" is probably not a valid PNG file`);
console.error('Unexpected PNG validation error', error);
```